include/transport/time_queue.h

```diff
@@ -314,13 +322,32 @@ namespace qtransport {
 
             const tick_type ticks = advance();
 
-            const auto future_bucket_index = (_bucket_index + (_duration / _interval) - 1) % _total_buckets;
+            const auto future_bucket_index = (_bucket_index + (_total_buckets - 1)) % _total_buckets;
+
+            internal_push(std::move(value), future_bucket_index, ticks + 1);
+        }
+
+        /**
+         * @brief Increment front
+         *
+         * @details This method should be called after front when the object is processed. This
+         *      will move the queue forward. If at the end of the queue, it'll be cleared and reset.
+         */
+         bool move_front() {
+            std::lock_guard<std::mutex> lock(_mutex);
+
+            if (_queue_index < _queue.size()) {
+                _queue_index++;
 
-            internal_push(std::move(value), _bucket_index + 1, ticks + 1);
+            } else if (_queue_index) {
+                _queue.clear();
+                _queue_index = 0;
+            }
```

Comment:
But front is already doing this? As is pop.

Reply:
That was one of the bugs.  Depending on time, front was actually popping a message that it shouldn't have been.  If we use front, what we really want to do is just move the index pointer when we want to remove the front object.